public/views/main.js

```diff
@@ -284,14 +284,14 @@ function onFrame(event, debug) {
 		}
 	}
 
-	// check for score change
+	// check for score change or score stayed at 999999 but the line count changed.
 	if (--pending_line === 0) {
 		if (
 			transformed.score
 			&& (transformed.lines != null)
-			&& (transformed.level != null)
-			&& diff.score >= 0
-			&& diff.cleared_lines >= 0
+			&& (transformed.level != null) 
+			&& ((diff.score >= 0 && diff.cleared_lines >= 0)
+			|| (transformed.score == 999999 && diff.cleared_lines > 0))
```

Comment:
Was this addition really needed? 🤔 . If `score` is stuck at 999,999, the `diff.score` will be zero, and clearing lines will be greater than 0, so it seems the previous conditional was enough, no?

Reply:
The reasoning for this change is the way I have implemented the score changing system is that it modifies the score saved in memory, but the transformed.score comes in every frame still at 999,999 even if the score is over that. This leads to a below 0 diff.score

I didn't know that about how the layouts are handled.
I will manually add this code to the player class for now, but once I have my head around this system some more I am happy to help with the porting of code to a unified solution.

Reply:
> https://github.com/timotheeg/nestrischamps

Oh right, calling `game.onLine(evt)` mutates the event object, so by the time it's stored in the last state of main, it's the incremented value of the score.

I'm actually thinking the event doesn't need to be mutated 🤔 . The score is displayed from the game data:
```
dom.score.current.textContent = game.data.score.current.toString().padStart(6, '0').padStart(7, ' ');
```
so as long as the data internal to the game instance is tracked correctly, the event and state can be treated as raw data, and the main file doesn't need to worry about the cap at 999,999. 

I think merge the PR first, but I might modify that later.


> I didn't know that about how the layouts are handled.

With dirty code and zero documentation, there was no way you could have known 😅 .


> I will manually add this code to the player class for now

Thank you! 🙏  I would have been happy with just main supporting this first, while implementation for player could be delayed. Thanks for jumping into it right away!